### PR TITLE
Migration tool fixes and enhancements

### PR DESCRIPTION
**Pull request by DavidePrincipi**
Refs
- https://github.com/NethServer/dev/issues/6782
- https://github.com/NethServer/dev/issues/6783

Additionally:
- do not run agent any more: it is not used at all
- change the storage of wg secret: make it persistent
- replace the node in the cluster, if already present
- avoid false ui alarms, if connection is down

### CODE REVIEW

File: root/etc/e-smith/db/configuration/migrate/delete_agent [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    # Disable the agent service
+    $DB->get_value_and_delete('agent') || return '';
+    system('systemctl', 'disable', '--now', 'agent');
```
Comment (gsanchietti) [inline root/etc/e-smith/db/configuration/migrate/delete_agent:4]: This is not needed: the system should stop the agent by itself.

File: root/etc/e-smith/db/configuration/migrate/wg_quick_conflict [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+    # Rename wg0 interface to ns8
+    my $oldRecord = $DB->get('wg-quick@wg0') || return '';
+    my $newRecord = $DB->new_record('wg-quick@ns8', { type => "service" });
+    $newRecord->merge_props($oldRecord->props);
+    $oldRecord->delete();
+    system('systemctl', 'disable', '--now', 'wg-quick@wg0');
```
Comment (gsanchietti) [inline root/etc/e-smith/db/configuration/migrate/wg_quick_conflict:7]: This should not be needed.

File: root/etc/e-smith/db/configuration/migrate/wg_quick_conflict [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    rename('/etc/wireguard/wg0.conf', '/etc/wireguard/ns8.conf');
+    system('systemctl', 'enable', '--now', 'wg-quick@ns8');
```
Comment (gsanchietti) [inline root/etc/e-smith/db/configuration/migrate/wg_quick_conflict:9]: This should not be needed: let's the system handle it

File: root/etc/e-smith/db/configuration/migrate/wg_quick_secretkey [hunk 4]
```diff
@@ -0,0 +1,12 @@
+{
+    use NethServer::Password;
+    my $secretKey = $DB->get_prop_and_delete('wg-quick@ns8', 'SecretKey');
+    if ($secretKey) {
+        my $tmpmask = umask 0077;
+        open(FH, ">", '/var/lib/nethserver/secrets/ns8wg');
+        print FH $secretKey . "\n";
+        close(FH);
+        umask $tmpmask;
+    }
```
Comment (gsanchietti) [inline root/etc/e-smith/db/configuration/migrate/wg_quick_secretkey:10]: What is the purpose of this change?

File: root/etc/e-smith/db/configuration/migrate/delete_agent (hunk 1 above)
Comment (gsanchietti) [reply root/etc/e-smith/db/configuration/migrate/delete_agent:4]: This is quite strange, esmith should check the default db value.
I'd still remove the disable call: only very few machines have this package installed.

File: ui/src/views/Dashboard.vue [hunk 5]
```diff
@@ -1265,6 +1265,7 @@ export default {
         },
         function(success) {
           context.loading.connectionUpdate = false;
+          context.error.connectionUpdate = "";
```
Comment (andre8244) [inline ui/src/views/Dashboard.vue:1268]: Same as above, please move this line to 1205

**Review by andre8244 — APPROVED** [review]

**Review by gsanchietti — APPROVED** [review]
